src/parser.rs

```diff
@@ -2022,16 +2130,31 @@ impl Parser {
         self.span_offset = span_position;
     }
 
-    pub fn lex_name(&mut self) -> Option<Token> {
+    /// More relaxed name lexing
+    pub fn lex_bareword(&mut self, name_strictness: NameStrictness) -> Option<Token> {
```

Comment:
Should "bareword" include keyword awareness?

Reply:
Yes, we should detect, e.g., if you're trying to do `def use ...`. We might do it after lexing, though, not sure. But in general, verifying valid names is envisioned to be a part of `naming.rs`.

Reply:
OK